graylog2-server/src/main/java/org/graylog2/rest/resources/system/ldap/LdapResource.java

```diff
@@ -209,15 +219,89 @@ public void updateLdapSettings(@ApiParam(name = "JSON body", required = true)
+    public Map<String, String> readGroupMapping(){
+        final LdapSettings ldapSettings = firstNonNull(ldapSettingsService.load(), ldapSettingsFactory.createEmpty());
+        return ldapSettings.getGroupMapping();
+    }
+
+    @PUT
+    @RequiresPermissions(value = {RestPermissions.LDAPGROUPS_EDIT, RestPermissions.LDAP_EDIT}, logical = OR)
+    @ApiOperation(value = "Update the LDAP group to Graylog role mapping", notes = "Corresponds directly to the output of GET /system/ldap/settings/groups")
+    @Path("/settings/groups")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response updateGroupMappingSettings(@ApiParam(name = "JSON body", required = true, value = "A hash in which the keys are the LDAP group names and values is the Graylog role name.")
+                                   @NotNull Map<String, String> groupMapping) throws ValidationException {
+        final LdapSettings ldapSettings = firstNonNull(ldapSettingsService.load(), ldapSettingsFactory.createEmpty());
+
+        ldapSettings.setGroupMapping(groupMapping);
+        ldapSettingsService.save(ldapSettings);
+
+        return Response.noContent().build();
+    }
+
+    @GET
+    @ApiOperation(value = "Get the available LDAP groups", notes = "")
+    @RequiresPermissions(RestPermissions.LDAPGROUPS_READ)
+    @Path("/groups")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Set<String> readGroups(){
+        final LdapSettings ldapSettings = firstNonNull(ldapSettingsService.load(), ldapSettingsFactory.createEmpty());
+
+        if (!ldapSettings.isEnabled()) {
+            throw new BadRequestException("LDAP is disabled.");
+        }
+        if (Strings.isNullOrEmpty(ldapSettings.getGroupSearchBase()) || Strings.isNullOrEmpty(ldapSettings.getGroupIdAttribute())) {
+            throw new BadRequestException("LDAP group configuration settings are not set.");
+        }
+
+        final LdapConnectionConfig config = new LdapConnectionConfig();
+        final URI ldapUri = ldapSettings.getUri();
+        config.setLdapHost(ldapUri.getHost());
+        config.setLdapPort(ldapUri.getPort());
+        config.setUseSsl(ldapUri.getScheme().startsWith("ldaps"));
+        config.setUseTls(ldapSettings.isUseStartTls());
+
+        if (ldapSettings.isTrustAllCertificates()) {
+            config.setTrustManagers(new TrustAllX509TrustManager());
+        }
+
+        if (!isNullOrEmpty(ldapSettings.getSystemUserName()) && !isNullOrEmpty(ldapSettings.getSystemPassword())) {
+            config.setName(ldapSettings.getSystemUserName());
+            config.setCredentials(ldapSettings.getSystemPassword());
+        }
+
+        try {
+            LdapNetworkConnection connection = ldapConnector.connect(config);
+            final Set<String> groups = ldapConnector.listGroups(connection,
+                                                               ldapSettings.getGroupSearchBase(),
+                                                               "group",
```

Comment:
yes, i haven't seen any other objectClass for ldap groups yet.
we'll adapt if someone files a bug :)